homeassistant/components/sensor/__init__.py

```diff
@@ -453,6 +478,74 @@ class SensorStateClass(StrEnum):
     SensorDeviceClass.WIND_SPEED: SpeedConverter,
 }
 
+DEVICE_CLASS_UNITS: dict[SensorDeviceClass, set[type[StrEnum] | str | None]] = {
```

Comment:
Do we need to have the same validation for number entities?
And would it make sense to have this dict shared somewhere?

Reply:
We have no validations on numbers whatsoever at this point. But yes, we can add more validation there too.

I suggest doing that at a later point